crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -478,28 +525,34 @@ impl BlobEventProcessor {
                     blob_sync_handler.clone(),
                     rx,
                     worker_index,
+                    cancel_token.clone(),
+                    node_cancel_token.clone(),
                 );
-                // TODO(WAL-876): gracefully shut down the background processor when the node is
-                // shutting down.
-                background_processors.push(Arc::new(tokio::spawn(async move {
+                background_processors.push(tokio::spawn(async move {
                     background_processor.run_background_event_processor().await;
-                })));
+                }));
```

Comment:
remark: in addition to propagating error back and stop the node, another thing we want to catch is run_background_event_processor panics. Although this can cause the node to stop when the next event sent to the worker is not successful, but it'll be nice if we can catch this earlier.